move token information into apt auth.conf on Debian 10...

rpasche opened the pull request:
## What does this PR change?
Since Debian 10, apt uses a netrc kind of file to hide the authentication
information that is needed to access repositories. It moves the token information
out of the "sources.list" file on Debian 10 systems into
new "auth.conf" files.

Warnings that are shown without the change

```
N: Usage of apt_auth.conf(5) should be preferred over embedding login information directly in the sources.list(5) entry for https://...
```

## GUI diff

No difference.

Before:

After:

- [X] **DONE**

## Documentation

- [ ] **DONE**

## Test coverage
- No tests: 

- [ ] **DONE**

## Links

Fixes #
Tracks # 

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

Shirocco88 commented:
see also https://github.com/uyuni-project/uyuni/issues/2523

Shirocco88 commented:
@rpasche: this issue also exist on Ubuntu-18.04 and Ubuntu-20.04 Systems;
please adapt the patch for fixing of Ubuntu-18.04 and Ubuntu-20.04 Systems.

Thanks in advance

Best regards
Shirocco88

rpasche commented:
I currently don't have a Ubuntu system to join into our uyuni. But if you tell me the is grains from Ubuntu, where this is needed, this should be easy to adjust

Robert

⁣sent from my mobile device​


-------- Originale Nachricht --------
Von: Shirocco88 <notifications@github.com>
Gesendet: Fri Sep 04 20:38:38 GMT+02:00 2020
An: uyuni-project/uyuni <uyuni@noreply.github.com>
Cc: Robert Paschedag <robert.paschedag@web.de>, Mention <mention@noreply.github.com>
Betreff: Re: [uyuni-project/uyuni] move token information into apt auth.conf on Debian 10... (#2572)

@rpasche: this issue also exist on Ubuntu-18.04 and Ubuntu-20.04 Systems;
please adapt the patch for fixing of Ubuntu-18.04 and Ubuntu-20.04 Systems.

Thanks in advance

Best regards
Shirocco88

-- 
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/uyuni-project/uyuni/pull/2572#issuecomment-687315796

Shirocco88 commented:
@rpasche: 
the grains of the operating systems with problems are:

```
=========================================
Ubuntu-18.04:
    os:
        Ubuntu
    os_family:
        Debian
    oscodename:
        bionic
    osfinger:
        Ubuntu-18.04
    osfullname:
        Ubuntu
    osmajorrelease:
        18
    osrelease:
        18.04
    osrelease_info:
        - 18
        - 4
 
=========================================
Ubuntu-20.04:
    os:
        Ubuntu
    os_family:
        Debian
    oscodename:
        focal
    osfinger:
        Ubuntu-20.04
    osfullname:
        Ubuntu
    osmajorrelease:
        20
    osrelease:
        20.04
    osrelease_info:
        - 20
        - 4
=========================================
Debian-10:
 
    os:
        Debian
    os_family:
        Debian
    oscodename:
        buster
    osfinger:
        Debian-10
    osfullname:
        Debian
    osmajorrelease:
        10
    osrelease:
        10
    osrelease_info:
        - 10
 
=========================================
```
 
Important: Debian-9 and Ubuntu-16.04 don't have problems.
 
Could you please also fix Issue #1266 because it is also relatet to OS-Grains and https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/channels/channels.repo
https://github.com/uyuni-project/uyuni/issues/1266
 
Thanks in advance.
 
Best regards
Shirocco88

rpasche commented:
> > > But if you tell me the is grains from Ubuntu, where this is needed, this should be easy to adjust
> 
> @rpasche:
> as requested, the grains of the operating systems with problems are:
> 
> ```
> =========================================
> Ubuntu-18.04:
>     os:
>         Ubuntu
>     os_family:
>         Debian
>     oscodename:
>         bionic
>     osfinger:
>         Ubuntu-18.04
>     osfullname:
>         Ubuntu
>     osmajorrelease:
>         18
>     osrelease:
>         18.04
>     osrelease_info:
>         - 18
>         - 4
>  
> =========================================
> Ubuntu-20.04:
>     os:
>         Ubuntu
>     os_family:
>         Debian
>     oscodename:
>         focal
>     osfinger:
>         Ubuntu-20.04
>     osfullname:
>         Ubuntu
>     osmajorrelease:
>         20
>     osrelease:
>         20.04
>     osrelease_info:
>         - 20
>         - 4
> =========================================
> Debian-10:
>  
>     os:
>         Debian
>     os_family:
>         Debian
>     oscodename:
>         buster
>     osfinger:
>         Debian-10
>     osfullname:
>         Debian
>     osmajorrelease:
>         10
>     osrelease:
>         10
>     osrelease_info:
>         - 10
>  
> =========================================
> ```
> 
> Important: Debian-9 and Ubuntu-16.04 don't have problems.
> 
> Could you please also fix Issue #1266 because it is also related to OS-Grains and https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/channels/channels.repo
> #1266
> 
> Thanks in advance.
> 
> Best regards
> Shirocco88

HI @Shirocco88 ,

sorry, but the fix for #1266...I did not add it. It seems it's not possible to use variables within those single quotes strings. So would need to be done somehow different. But I added the change for Ubuntu. I tested the changes on my Uyuni system at least with SLES, Debian 9 and Debian 10....and it is looking good so far. As said, I do not have Ubuntu systems....so I cannot finally test this.

Cheers
Robert

Shirocco88 commented:
>> I added the change for Ubuntu.
@rpasche:: Thanks!!! 
I will test it, if merged into master

Best regards
Shirocco88

rpasche commented:
> > > I added the change for Ubuntu.
> 
> @rpasche:: Thanks!!!
> I will test it, if merged into master
> 
> Best regards
> Shirocco88

Wait....saw an issue with the aptauth.conf file. There is an unneeded change if os

> > > I added the change for Ubuntu.
> 
> @rpasche:: Thanks!!!
> I will test it, if merged into master
> 
> Best regards
> Shirocco88

Hold on a second. Just saw an unneeded change within aptauth.conf. As we check for ubuntu and Debian within the `init.sls`, that check is unnecessary. Will remove it.

rpasche commented:
> > > > I added the change for Ubuntu.
> > 
> > 
> > @rpasche:: Thanks!!!
> > I will test it, if merged into master
> > Best regards
> > Shirocco88
> 
> Wait....saw an issue with the aptauth.conf file. There is an unneeded change if os
> 
> > > > I added the change for Ubuntu.
> > 
> > 
> > @rpasche:: Thanks!!!
> > I will test it, if merged into master
> > Best regards
> > Shirocco88
> 
> Hold on a second. Just saw an unneeded change within aptauth.conf. As we check for ubuntu and Debian within the `init.sls`, that check is unnecessary. Will remove it.

Done.

mateiw reviewed (changes requested):
Thanks for the patch @rpasche. Just one small thing, otherwise it looks good.

rpasche commented on susemanager-utils/susemanager-sls/salt/channels/init.sls:
```diff
@@ -72,6 +72,18 @@ mgrchannels_repo:
+{%- if (grains['os'] == 'Debian' and grains['osmajorrelease']|int >= 10) or (grains['os'] == 'Ubuntu' and grains['osmajorrelease']|int >= 18) %}
+aptauth_conf:
+  file.managed:
+    - name: "/etc/apt/auth.conf.d/susemanager.conf"
```
@mateiw Updated branch. Only added to `cleanup_minion` as that is also included within `cleanup_ssh_minion`

mateiw reviewed (approved):
LGTM :+1:

mateiw commented:
Merged. Thanks for the contribution @rpasche.
